modules/ROOT/pages/java-support.adoc

```diff
@@ -0,0 +1,160 @@
+
+Java 17 upgrade planning includes evaluating integration apps, connectors (Anypoint connectors and custom connectors), Mule Gateway policies, and API proxies, as all of these components rely on Java. 
+
+[[mule-runtime-java-support]]
+== Mule Runtime Java Support
+
+The following table shows the timeline for Java 8, Java 11, and Java 17 support:
+
+[%header%autowidth.spread]
+|===
+|Mule Version| Release Date |Java Version | End of Standard Support | End of Extended Support
+|4.5 Edge | October 2023 |8 and 11 | February 2024 | June 2024
+|4.6 LTS | February 2024 | 8, 11, and 17 | February 2025 | February 2026
+|4.6 Edge | February 2024 |8, 11, and 17 | June 2024 | October 2024
+|4.7 Edge | June 2024 | 8, 11, and 17 | October 2024 | February 2025
+|4.8 Edge | October 2024 | 8, 11, and 17 | February 2025 | June 2025
+|4.9 LTS | February 2025 | 17 | February 2026 | February 2027
+|4.9 Edge | February 2025 | 17 | June 2025 | October 2025
+|===
+
+[NOTE]
+Standard support for Java 8 and 11 ends with the Mule runtime 4.9 release in February 2025, so plan your upgrade path for apps that are running on either Java 8 or Java 11 accordingly.
+
+For more information about Mule release cadences, see xref:release-notes::mule-runtime/lts-edge-release-cadence.adoc[].
+
+[[mule-runtime]]
+== Mule Runtime
+
+To take advantage of the improved performance and security of Java 17, xref:release-notes::mule-runtime/updating-mule-4-versions.adoc[upgrade to Mule runtime 4.6 or later].
+
+Follow these best practices:
+
+* Upgrade your standalone Mule instances.
+* Upgrade on-premises Mule instances that are managed through Runtime Manager.
+* Update the Mule version used to build your Mule applications.
+* xref:mule-runtime::deploying.adoc[Deploy your applications] to a target running on Mule 4.6 or later.
+
+[[mule-apps]]
+== Mule Apps
+
+The majority of Mule applications currently in use can be updated, tested, and redeployed. However, any apps and connectors that are running custom Java code require additional work to certify those components. Before you upgrade your integration apps or Mule Gateway policies and proxies to Java 17, all extensions, modules, and connectors used within those apps and policies must first be updated to be Java 17-compatible. 
+
+[IMPORTANT]
+To ensure your API proxies or Mule apps are protected when upgrading, upgrade your API policies *before* upgrading your API proxies or Mule apps.
+
+[[application-performance]]
+=== Application Performance
+
+Java 17 manages memory differently than it did in earlier versions like Java 8. In some cases, Java 17 preferentially commits to heap memory or uses heap memory for structures in some apps, which earlier Java versions allocated to native memory.
+
+Because of this change in Java 17 memory management, it is important for you to conduct xref:mule-runtime::tuning-test-validations.adoc[performance testing] and xref:mule-runtime::tuning-monitoring.adoc[monitoring] for your most common workloads.
```

Comment:
```suggestion
Because of this change in Java 17 memory management, you must conduct xref:mule-runtime::tuning-test-validations.adoc[performance testing] and xref:mule-runtime::tuning-monitoring.adoc[monitoring] for your most common workloads.
```